libraries/Microsoft.Bot.Builder/Teams/TeamsInfo.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Teams;
+using Microsoft.Bot.Schema.Teams;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Teams
+{
+    public static class TeamsInfo
+    {
+        public static async Task<TeamDetails> GetTeamDetailsAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
+        {
+            var teamId = GetTeamId(turnContext) ?? throw new InvalidOperationException("This method is only valid within the scope of MS Teams Team.");
+            return await GetTeamsConnectorClient(turnContext).Teams.FetchTeamDetailsAsync(teamId, cancellationToken).ConfigureAwait(false);
+        }
+
+        public static async Task<IList<ChannelInfo>> GetChannelsAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
+        {
+            var teamId = GetTeamId(turnContext) ?? throw new InvalidOperationException("This method is only valid within the scope of MS Teams Team.");
+            var channelList = await GetTeamsConnectorClient(turnContext).Teams.FetchChannelListAsync(teamId, cancellationToken).ConfigureAwait(false);
+            return channelList.Conversations;
```

Comment:
Seems silly that it returns a class with one property... I supposed the [ConversationList](https://github.com/microsoft/botbuilder-dotnet/blob/d8ecbd83addd6ba8d1cdade41e5b8298f5cf3809/libraries/Microsoft.Bot.Schema/Teams/Generated/ConversationList.cs) class could be considered extensible.

That said, I think you need to rename the [extension method](https://github.com/microsoft/botbuilder-dotnet/blob/d8ecbd83addd6ba8d1cdade41e5b8298f5cf3809/libraries/Microsoft.Bot.Connector/Teams/TeamsOperationsExtensions.cs#L39).

Reply:
This is the Teams generated code from the swagger. The naming is slightly confusing but its pretty buried here.